jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -109,6 +109,13 @@
     JERRY_ASSERT (utf8_ptr != NULL); \
     jmem_heap_free_block ((void *) utf8_ptr, utf8_str_size); \
   }
+/**
+ * This is very ugly macro but there is not better solution,
+ * which will work for bit fields in versions of GCC used in MCU-s
+ * See:
+ * https://gcc.gnu.org/bugzilla/show_bug.cgi?id=39170
+ */
+#define ECMA_CONVERT_BOOL_TO_FIELD(x) ((x) ? 1 : 0)
```

Comment:
Please make this dependent from the GCC version and declare as empty above the problematic version. Do not criticize your own code in the comment, just write that it is a workaround for a certain GCC version.

Reply:
I think compilers are able to optimize this, so it si not an ugly macro.